public/app/features/trails/MetricSelect/AddToExplorationsButton.tsx

```diff
@@ -0,0 +1,139 @@
+export const addToExplorationsButtonLabel = 'add panel to exploration';
+
+export interface AddToExplorationButtonState extends SceneObjectState {
+  frame?: DataFrame;
+  dsUid?: string;
+  labelName?: string;
+  fieldName?: string;
+  context?: ExtensionContext;
+
+  disabledLinks: string[];
+  queries: DataQuery[];
+}
+
+interface ExtensionContext {
+  timeRange: TimeRange;
+  queries: DataQuery[];
+  datasource: DataSourceRef;
+  origin: string;
+  url: string;
+  type: string;
+  title: string;
+  id: string;
+  logoPath: string;
+  note?: string;
+  drillDownLabel?: string;
+}
+
+export class AddToExplorationButton extends SceneObjectBase<AddToExplorationButtonState> {
+  constructor(state: Omit<AddToExplorationButtonState, 'disabledLinks' | 'queries'>) {
+    super({ ...state, disabledLinks: [], queries: [] });
+
+    this.addActivationHandler(this._onActivate.bind(this));
+  }
+
+  private _onActivate = () => {
+    this._subs.add(
+      this.subscribeToState(() => {
+        this.getQueries();
+        this.getContext();
+      })
+    );
+
+    const datasourceUid = sceneGraph.interpolate(this, VAR_DATASOURCE_EXPR);
+    this.setState({ dsUid: datasourceUid });
+  };
+
+  private readonly getQueries = () => {
+    const data = sceneGraph.getData(this);
+    const queryRunner = sceneGraph.findObject(data, isQueryRunner);
+
+    if (isQueryRunner(queryRunner)) {
+      const filter = this.state.frame ? getFilter(this.state.frame) : null;
+      const queries = queryRunner.state.queries.map((q) => ({
+        ...q,
+        expr: sceneGraph.interpolate(queryRunner, q.expr),
+        legendFormat: filter?.name ? `{{ ${filter.name} }}` : sceneGraph.interpolate(queryRunner, q.legendFormat),
+      }));
+
+      if (JSON.stringify(queries) !== JSON.stringify(this.state.queries)) {
+        this.setState({ queries });
+      }
+    }
+  };
+
+  private readonly getContext = () => {
+    const { queries, dsUid, labelName, fieldName } = this.state;
+    const timeRange = sceneGraph.getTimeRange(this);
+
+    if (!timeRange || !queries || !dsUid) {
+      return;
+    }
+    const ctx = {
+      origin: 'Explore Metrics',
+      type: 'timeseries',
+      queries,
+      timeRange: { ...timeRange.state.value },
+      datasource: { uid: dsUid },
+      url: window.location.href,
+      id: `${JSON.stringify(queries)}${labelName}${fieldName}`,
+      title: `${labelName}${fieldName ? ` > ${fieldName}` : ''}`,
+      logoPath: MimirLogo,
+      drillDownLabel: fieldName,
+    };
+    if (JSON.stringify(ctx) !== JSON.stringify(this.state.context)) {
+      this.setState({ context: ctx });
+    }
+  };
+
+  public static Component = ({ model }: SceneComponentProps<AddToExplorationButton>) => {
+    const { context, disabledLinks } = model.useState();
+    const { links } = usePluginLinks({ extensionPointId, context, limitPerPlugin: 1 });
+    const link = links.find((link) => link.pluginId === explorationsPluginId);
+
+    if (!link) {
```

Comment:
This part confused me when I had not added the sidecar feature toggle. Would it be possible to wrap the component in this and add some comments as documentation? 

```
// This feature is part of the explorations app. Here is a link to the repo. The repo has instructions for building the app and the necessary feature toggles. LINK
```

But I can see how having the check inside the component is better for tests.